packages/shared/src/components/cards/ActionButtons.tsx

```diff
@@ -103,6 +104,18 @@ export default function ActionButtons({
     </>
   );
 
+  const isSharedPostSquadPost =
+    post.sharedPost?.source.type === SourceType.Squad;
+
+  const insaneReadArticleLink = () => {
+    if (post.type === PostType.Share) {
+      return isSharedPostSquadPost
+        ? post.sharedPost.commentsPermalink
+        : post.sharedPost.permalink;
+    }
+    return post.permalink;
+  };
```

Comment:
This looks like something that we can have as util and then use it with arguments vs having an inline function in component. `getReadArticleLink({ post })`